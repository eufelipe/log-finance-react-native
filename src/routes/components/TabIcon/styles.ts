import styled from 'styled-components/native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import shadow from 'styles/shadow';

export const Icon = styled(MaterialIcons).attrs({
  size: 32,
})``;

export const AddButtonContainer = styled.View`
  background-color: ${({theme}) => theme.colors.seconday};
  padding: ${({theme}) => theme.spacing.tall}px;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: -60px;
  align-self: center;
  z-index: 100;
  border-radius: 50px;
  elevation: 20;
  ${shadow}
`;
