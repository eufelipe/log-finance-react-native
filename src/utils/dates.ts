import {
  format as formatPattern,
  isValid,
  parseISO as parseISOFNS,
} from 'date-fns';

import locale from 'date-fns/locale/pt-BR';

export const DATE_FORMAT_ISO = 'yyyy-MM-dd';
export const DATE_FORMAT_BR = 'dd/MM/yyyy';
export const DATE_FORMAT_FRIENDLY = " dd/MM 'às' HH:mm";
export const DATE_FORMAT_FRIENDLY_TIME = "'às' HH:mm";

export function parseStringToDate(value: string): Date {
  const parsedDate = parseISOFNS(value);
  if (!isValid(parsedDate)) {
    throw Error('canot parse date');
  }
  return parsedDate;
}
export function parseDateToString({
  value,
  format = DATE_FORMAT_BR,
}: {
  value: Date;
  format?: string;
}): string {
  return formatPattern(value, format, {locale});
}

export const getDate = (date: Date): string => {
  return formatPattern(date, DATE_FORMAT_ISO, {locale});
};

export const getDateToday = (): string => {
  const today = new Date();
  return formatPattern(today, DATE_FORMAT_ISO, {locale});
};

export const getDateAndHourFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY, {locale});
};

export const getTimeFriendly = (value: Date): string => {
  return formatPattern(value, DATE_FORMAT_FRIENDLY_TIME, {locale});
};
