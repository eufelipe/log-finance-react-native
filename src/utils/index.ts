export {default as number} from './number';
export {default as strings} from './strings';
