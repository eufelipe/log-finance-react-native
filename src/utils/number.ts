export const sanitizeToFloat = (value: string): number => {
  value = value.replace(/[^0-9.]+/, '');
  if (value.includes('.') && value.includes(',')) {
    value = value.replace(/\./g, '').replace(',', '.');
  } else if (value.includes(',')) {
    value = value.replace(',', '.');
  }
  return parseFloat(value) || 0;
};
