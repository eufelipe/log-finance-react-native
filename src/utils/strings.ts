export const sanitizeString = (value: string): string => {
  return value
    .toLowerCase()
    .replace(/á/g, 'a')
    .replace(/é/g, 'e')
    .replace(/í/g, 'i')
    .replace(/ó/g, 'o')
    .replace(/ú/g, 'u');
};
