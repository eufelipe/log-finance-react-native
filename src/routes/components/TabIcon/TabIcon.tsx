import React from 'react';

import {
  TAB_KEY_HOME,
  TAB_KEY_ADD_ENTRY,
  TAB_KEY_REPORTS,
} from 'routes/TabsRoute';

import Colors from 'styles/colors';

import {Icon, AddButtonContainer} from './styles';

interface Props {
  route: string;
  focused?: boolean;
}

const TabIcon = ({route, focused}: Props): JSX.Element => {
  const icons = {
    [`${TAB_KEY_HOME}`]: 'home',
    [`${TAB_KEY_ADD_ENTRY}`]: 'add',
    [`${TAB_KEY_REPORTS}`]: 'assessment',
  };

  const icon = icons[route] ?? 'crop-square';

  if (route === TAB_KEY_ADD_ENTRY) {
    return (
      <AddButtonContainer>
        <Icon name={icon} color={Colors.white} />
      </AddButtonContainer>
    );
  }

  return <Icon name={icon} color={focused ? Colors.white : Colors.gray} />;
};

export default TabIcon;
