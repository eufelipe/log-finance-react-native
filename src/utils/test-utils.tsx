/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {render, RenderAPI, RenderOptions} from '@testing-library/react-native';

import {theme} from 'styles/theme';
import {ThemeProvider} from 'styled-components/native';

import {NavigationContainer} from '@react-navigation/native';

interface Props {
  children: JSX.Element;
}

const AllTheProviders = ({children}: Props) => {
  return (
    <ThemeProvider theme={theme}>
      <NavigationContainer>{children}</NavigationContainer>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions,
): RenderAPI => render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
