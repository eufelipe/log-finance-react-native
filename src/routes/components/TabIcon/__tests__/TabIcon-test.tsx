/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';

import {render} from 'utils/test-utils';

import TabIcon from '../TabIcon';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

describe('TabIcon', () => {
  it('Test match snapshot TabIcon', () => {
    // given
    const props = {
      route: 'Home',
      focused: true,
    };

    // when
    const {toJSON} = render(<TabIcon {...props} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
