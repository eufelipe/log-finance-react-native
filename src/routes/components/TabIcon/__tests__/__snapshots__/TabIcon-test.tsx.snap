// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabIcon Test match snapshot TabIcon 1`] = `
<Icon
  color="#ffffff"
  name="home"
  size={32}
  style={
    Array [
      Object {},
    ]
  }
/>
`;
